feat/feature-flags.json

```diff
@@ -0,0 +1,17 @@
+{
```

Comment:
🤔 How should this file be managed across multiple environments?

Reply:
Would we name the file after the environment?

`local.json`, `staging.json`, `production.json`?

Reply:
We could do that, but then the JSON import will need to be dynamic. Not sure how that'll affect TS infererence.

Reply:
That stinks! I wonder if we could update the `defaultState` and `status` keys to be objects that define it per environment, defaulting to being the same for all environments if they're not an object?

Something like:

```
"favorite_feature": {
  "status": {
    "local": "enabled",
    "staging": "switchable",
    "production": "disabled",
  }
}
```

We'd have to define a cascading behavior though, like if a lower environment isn't defined then it uses the definition for the next highest environment? I.e., if local isn't there, then use staging, if not then production (at which point it shouldn't be the object version anyway). And if we're on a higher environment than is defined, default to `disabled`.

Do you think that could work Dhruv? It makes things a bit more complicated but it would give us a lot more flexibility.

Reply:
I'll explore this and see if we can work with that.